apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -318,6 +319,13 @@ func (r *OpenStackControlPlane) Default() {
 	r.DefaultServices()
 }
 
+// Helper function to initialize overrideSpec object. Could be moved to lib-common.
+func initializeOverride(override **route.OverrideSpec) {
```

Comment:
I thought it would also receive route annotations as arg, then it would

override.Route.AddAnnotations(annotations)...

otherwise this helper is very bare and probably is not very helpful

Reply:
Oooh I see... I have changed it now to include that. Let me know if it looks better. Functional test was also added.